scene/main/viewport.cpp

```diff
@@ -2015,7 +2015,7 @@ void Viewport::_gui_input_event(Ref<InputEvent> p_event) {
 		if (!gui.drag_attempted && gui.mouse_focus && section_root && !section_root->gui.global_dragging && (mm->get_button_mask().has_flag(MouseButtonMask::LEFT))) {
 			gui.drag_accum += mm->get_relative();
 			float len = gui.drag_accum.length();
-			if (len > 10) {
+			if (len > (Engine::get_singleton()->is_editor_hint() ? 10 : gui.drag_threshold)) {
```

Comment:
```suggestion
			if (len > gui.drag_threshold) {
```

This editor hint check should not be necessary, `sml->get_root()` will only set it for running project, so editor will use default value in any case. And there's no reason it can't be changed for a specific viewport in the editor in the future.